app/js/petition.js

```diff
@@ -0,0 +1,155 @@
+      VALID=false;
+      alert("The city field is invalid! It must only have characters and spaces.");
+    } else if (!stringValidate(province)) {
+      VALID=false;
+      alert("The province field is invalid! It must only have characters and spaces.");
+    } else if (!postalValidate(postalcode)) {
+      VALID=false;
+      alert("The postal code field is invalid! It must only have characters, numbers and spaces.");
+    } else if (!duplicateValidate(SUBMITTEDDATA, PASTDATA)) {
+      VALID = false;
+      alert("Your information is a duplicate.")
+    }
+
+    //Validation Functions - Returns true when input is valid
+    function stringValidate(str) {
+      var regex = /[^a-z,\s]/gi;
+      return !regex.test(str);
+    }
+    function stringAndNumberValidate(str) {
+      var regex = /[^a-z,\s,\d,\-,\.,\@,\/]/gi;
+      return !regex.test(str);
+    }
+    function postalValidate(str) {
+      var regex = /[^a-z,\d]/gi;
+      str = str.replace(" ", "")
+      return !(regex.test(str) && (str.length == 6));
+    }
+    function duplicateValidate(arrs, arrp) {
+      for (var i = 0; i < arrp.length; i++) {
+        if (arrp[i][0] == arrs[0] && arrp[i][1] == arrs[1] && arrp[i][2] == arrs[2] && arrp[i][3] == arrs[3] && arrp[i][4] == arrs[4] && arrp[i][5] == arrs[5]) {
+          return false;
+        }
+      }
+      return true;
+    }
+    //Submit form with ajax if valid
+    if (VALID) {
+      submitForm();
+    }
+    function newData(arrs, arrp) {
+      return arrp.push(arrs);
+    }
+    function submitForm() {
+        $.ajax({
+                  url: "https://docs.google.com/forms/d/e/1FAIpQLScMawBqg4oxvJVarWwRPjDpqtSu5MUdNq_dBgphyruz5e8z0g/formResponse",
+                  data: {"entry.843751608" : SUBMITTEDDATA[0],
+                         "entry.786666914" : SUBMITTEDDATA[1],
+                         "entry.113893232" : SUBMITTEDDATA[2],
+                         "entry.475910473" : SUBMITTEDDATA[3],
+                         "entry.1430871484": SUBMITTEDDATA[4],
+                         "entry.2000257846": SUBMITTEDDATA[5]},
+                  type: "POST",
+                  dataType: "xml",
+                  statusCode: {
+                      0: function (){
+                        alert("Thank you for signing the petition.")
+                      },
+                      200: function (){
+                        alert("Thank you for signing the petition.")
+                      },
+                      //400 - Bad Request
+                      400: function(){
+                        alert("There was error within the server. Please try again at later time.");
+                      }
+                      ,
+                      //404 - Not Found
+                      404: function(){
+                        alert("There was error with the form. Please try again at later time.");
+                      }
+                  }
+              });
+        location.reload();
+    };
```

Comment:
Duplicate validation doesn't seem to work with the semi-colon. Don't ask me why. lol